anvil/src/eth/backend/mem/mod.rs

```diff
@@ -440,10 +455,15 @@ impl Backend {
     }
 
     /// Executes the `CallRequest` without writing to the DB
+    ///
+    /// **Note:**: in order to make it consistent this accepts a `Option<BlockNumber>`, it's
+    /// expected that this number is in range, See [Self::ensure_block_number()].  If the block
+    /// number is out of range, then we pick the latest block
```

Comment:
Should we be throwing an error instead?

Reply:
was kinda on the fence about this, but now that I'm having another look, making this function return a Result instead would be better even if it's a bit more verbose, but this is only called from functions that return a result themselves, so it's fine.